language-plutus-core/src/Language/PlutusCore/Lexer/Type.hs

```diff
@@ -51,6 +55,19 @@ data BuiltinName = AddInteger
                  | BlockNum
                  deriving (Show, Eq, Ord, Enum, Bounded, Generic, NFData, Lift)
 
+-- | The type of dynamic builtin functions. I.e. functions that exist on certain chains and do
+-- not exist on others. Each 'DynamicBuiltinName' has an associated type and operational semantics --
+-- this allows to type check and evaluate dynamic builtins just like static ones.
+newtype DynamicBuiltinName = DynamicBuiltinName
+    { unDynamicBuiltinName :: T.Text  -- ^ The name of a dynamic builtin function.
```

Comment:
Do we want a `ShortText` or the like? We would ideally like dynamic builtin names to be cheap to compare.

Reply:
Yes, but

1. I tried using `ShortText`. A lof of instances are missing and it's inconvenient
2. wer're going to implement the more heavy solution with another type variable, so this is just an intermediate solution and `Text` is good enough